packages/components/src/components/carousel/carousel-utils.ts

```diff
@@ -65,11 +65,13 @@ export const warnIfHeadingIsMissing = (host: HTMLElement, heading: string): void
   }
 };
 
-export const getSlidesAndAddNamedSlots = (host: HTMLElement): HTMLElement[] => {
+export const getSlidesAndAddAttributes = (host: HTMLElement): HTMLElement[] => {
   const slides = Array.from(host.children).filter(
     ({ slot }) => slot !== 'heading' && slot !== 'description'
   ) as HTMLElement[];
-  slides.forEach((el, i) => el.setAttribute('slot', `slide-${i}`));
+  slides.forEach((el, i) => {
+    Object.assign(el, { slot: `slide-${i}`, tabIndex: 0 });
```

Comment:
Does this work cross browser since it implies properties are synced to attributes?

Reply:
Hm, good point. So we should do it better like this?

```slides.forEach((el, i) => {
    el.setAttribute('slot', `slide-${i}`);
    el.setAttribute('tabindex', '0');
  });```

Reply:
That would be the safer option, yes. But unit tests should also verify this behavior because now they check that the properties are set which might happen in jsdom but not in real browser.